### PR TITLE
TASK: Provide alternative for `NodeDiscriminator::fromNode` that doesnt use Nodes content stream

### PR DESCRIPTION
**Pull request by mhsdesign**
Related #5034 and #5043

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by mhsdesign** [comment]
I think this is a valid way, question is do we want to do it like this for the rest of the usages of `NodeDiscriminator::fromNode` as well (ca 8)?

cc @nezaniel

**Comment by mhsdesign** [comment]
Okay so i tried 3 different ways of achieving the goal (not to use $node->subgraphIdentity->contentStreamId) and im now kinda happy with the last attempt of introducing a `ContentStreamAwareNode` (+ builder) for the tests.

Horray 🥳

File: Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/Helpers/ContentStreamAwareNode.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+
+final readonly class ContentStreamAwareNode
+{
+    private function __construct(
+        public ContentStreamId $contentStreamId,
+        public Node $nodeInstance,
+        /** Alias for $currentNode->instance->aggregateId */
+        public NodeAggregateId $aggregateId,
```
Comment (mhsdesign) [inline Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/Helpers/ContentStreamAwareNode.php:17]: that way we dont need to rewrite too much code ...

File: Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/ProjectedNodeTrait.php [hunk 2]
```diff
@@ -549,17 +563,17 @@ public function iExpectThisNodeToNotBeReferenced(): void
...
     {
         $expectedParentDiscriminator = NodeDiscriminator::fromShorthand($serializedParentNodeDiscriminator);
-        $this->assertOnCurrentNode(function (Node $currentNode) use ($expectedParentDiscriminator) {
+        $this->assertOnCurrentContentStreamAwareNode(function (ContentStreamAwareNode $currentNode) use ($expectedParentDiscriminator) {
             $subgraph = $this->getCurrentSubgraph();
 
             $parent = $subgraph->findParentNode($currentNode->aggregateId);
             Assert::assertInstanceOf(Node::class, $parent, 'Parent not found.');
-            $actualParentDiscriminator = NodeDiscriminator::fromNode($parent);
+            $actualParentDiscriminator = NodeDiscriminator::fromNode($currentNode->builder()->buildNode($parent));
```
Comment (mhsdesign) [inline Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/ProjectedNodeTrait.php:571]: alternative: ... if we dislike the god object

```suggestion
            $actualParentDiscriminator = NodeDiscriminator::fromNode(ContentStreamAwareNodeBuilder::fromNode($currentNode)->buildNode($parent));
```

File: Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/Helpers/ContentStreamAwareNode.php
Comment (bwaidelich) [inline Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/Helpers/ContentStreamAwareNode.php:1]: We agreed not to have this kind of DTOs today (see https://github.com/neos/neos-development-collection/issues/5034#issuecomment-2196640446)

**Comment by mhsdesign** [comment]
Closing in favour of https://github.com/neos/neos-development-collection/pull/5169

**Pull request closed without merging**